Entity/BaseUser.php

```diff
@@ -232,4 +235,26 @@ public function canSwitchTo(BaseUser $user): bool
     {
         return false;
     }
+
+    public function serialize(): string
+    {
+        return serialize(
+            [
+                $this->id,
+                $this->username,
+                $this->password,
+                $this->salt,
+            ]
+        );
+    }
+
+    public function unserialize($serialized): void
+    {
+        [$this->id, $this->username, $this->password, $this->salt] = unserialize($serialized);
+    }
+
+    public function isEqualTo(UserInterface $user): bool
+    {
+        return $user->getUsername() === $this->getUsername();
```

Comment:
Why not with id? Because some apps we use email address as unique value instead of username

Reply:
@jonasdekeukelaere I changed this to compare the ids. The problem there was that `UserInterface` does not have a `getId()` method, only a `getUsername()` (and no `getEmailAddress()` which seems logic to me).